### PR TITLE
fix email pattern: comply with RFC5322

### PR DESCRIPTION
**Pull request by lemontree210**

### CODE REVIEW

**Review by Roxe322 — APPROVED** [review]

File: samanthas_telegram_bot/data_structures/constants.py [hunk 1]
```diff
@@ -7,7 +7,10 @@
 #  Maybe this can be changed anyway (not necessarily in MVP).
 #  We could check ID's of phrases at the start to make sure there's no mismatch.
 
-EMAIL_PATTERN = re.compile(r"^([\w\-.]+)@([\w\-.]+)\.([a-zA-Z]{2,5})$")
+EMAIL_PATTERN = re.compile(
+    r"^([-!#-'*+/-9=?A-Z^-~]+(\.[-!#-'*+/-9=?A-Z^-~]+)*|\"([]!#-[^-~ \t]|(\\[\t -~]))+\")"
+    r"@([-!#-'*+/-9=?A-Z^-~]+(\.[-!#-'*+/-9=?A-Z^-~]+)*|\[[\t -Z^-~]*])$"
```
[the remark this answers is not in the document]
Comment (lemontree210) [reply samanthas_telegram_bot/data_structures/constants.py]: Fixed

[the remark this answers is not in the document]
Comment (lemontree210) [reply samanthas_telegram_bot/data_structures/constants.py]: Fixed

[the remark this answers is not in the document]
Comment (lemontree210) [reply samanthas_telegram_bot/data_structures/constants.py]: Fixed

[the remark this answers is not in the document]
Comment (lemontree210) [reply samanthas_telegram_bot/data_structures/constants.py]: Fixed

**Pull request merged**